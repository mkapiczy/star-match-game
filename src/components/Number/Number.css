.number {
    margin-right: 5px;
    background-color: #aeaeae;
    border: none;
    color: white;
    text-align: center;
    text-decoration: none;
    border-radius: 50%;
    width: 5vmin;
    height: 5vmin;
}

.number.not-available {
    background-color: #3F51B5;
}

.number:not(.not-available):hover {
    background-color: #d7d7d7;
}