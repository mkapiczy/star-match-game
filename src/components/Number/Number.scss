.number {
  margin-right: 0.25em;
  margin-left: 0.25em;
  background-color: #aeaeae;
  border: none;
  color: white;
  text-align: center;
  text-decoration: none;
  border-radius: 50%;
  width: 5vmin;
  height: 5vmin;
  font-size: 2.5vmin;

  &.not-available {
    background-color: #3F51B5;
  }

  &:not(.not-available):hover {
    background-color: #d8d8d8;
  }
}