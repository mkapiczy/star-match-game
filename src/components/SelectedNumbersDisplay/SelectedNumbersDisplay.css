.SelectedNumbersDisplay {
    flex: 1 0 auto;
    background-color: black;
}
