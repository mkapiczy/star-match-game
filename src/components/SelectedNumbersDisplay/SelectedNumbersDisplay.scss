.SelectedNumbersDisplay {
    display: flex;
    flex-direction: row;
    flex: 3 0 auto;
    flex-wrap: wrap;
    width: 33.333%;
    height: 80%;
    background-color: white;
    justify-content: flex-start;
    align-items: center;
    border-radius: 2.5vmin;
    border-style: groove;
    margin: 2vmin;
}
