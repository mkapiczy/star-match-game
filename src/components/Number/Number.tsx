import React from 'react';
import './Number.scss';
import classNames from "classnames";

interface Props {
    numberValue: number,
    onNumberClick: (numberValue: number) => void;
    isAvailable: boolean
}

const Number: React.FC<Props> = (props: Props) => {
    return (
        <button key={props.numberValue} className={classNames('number', {'not-available': !props.isAvailable})}
                disabled={!props.isAvailable} onClick={() => props.onNumberClick(props.numberValue)}>
            {props.numberValue}
        </button>
    )
}

export default Number;