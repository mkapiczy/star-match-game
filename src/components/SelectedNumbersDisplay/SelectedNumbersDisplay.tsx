import React from 'react';
import './SelectedNumbersDisplay.scss';
import Number from '../Number/Number'

interface Props {
    selectedNumbers: Array<number>
    onNumberClick: (selectedNumber: number) => void;
}

const SelectedNumbersDisplay: React.FC<Props> = (props: Props) => {
    return (
        <div className="SelectedNumbersDisplay">
            {props.selectedNumbers.map(number => <Number key={number} numberValue={number} onNumberClick={props.onNumberClick} isAvailable={true}/>)}
        </div>
    )
}


export default SelectedNumbersDisplay;