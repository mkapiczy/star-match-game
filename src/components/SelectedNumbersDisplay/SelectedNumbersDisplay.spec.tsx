import React from 'react';
import SelectedNumbersDisplay from "./SelectedNumbersDisplay";
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import Number from "../Number/Number";

describe("SelectedNumbersDisplay component snapshot tests", () => {
    it("should match snapshot for all numbers selected", () => {
        const component = renderer.create(
            <SelectedNumbersDisplay onNumberClick={() => ''} selectedNumbers={[1, 2, 3, 4, 5, 6, 7, 8, 9]}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("should match snapshot for no numbers selected", () => {
        const component = renderer.create(
            <SelectedNumbersDisplay onNumberClick={() => ''} selectedNumbers={[]}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("should match snapshot for some numbers selected", () => {
        const component = renderer.create(
            <SelectedNumbersDisplay onNumberClick={() => ''} selectedNumbers={[1, 2, 7, 8]}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('SelectedNumbersDisplay component tests', () => {
    it("should render correct number of Number components", () => {
        const selectedNumbers = [1, 2, 3, 4, 5, 6, 7, 8, 9]
        const wrapper = mount(<SelectedNumbersDisplay onNumberClick={() => ''}
                                                      selectedNumbers={selectedNumbers}/>)
        const renderedNumberComponents = wrapper.find(Number)
        expect(renderedNumberComponents.length).toBe(selectedNumbers.length)
    })
    it("should render correct numbers values", () => {
        const selectedNumbers = [1, 2, 3, 4, 5, 6, 7, 8, 9]
        const wrapper = mount(<SelectedNumbersDisplay onNumberClick={() => ''}
                                                      selectedNumbers={selectedNumbers}/>)
        const renderedNumberComponents = wrapper.find(Number)
        expect(renderedNumberComponents.length).toBe(selectedNumbers.length)
        renderedNumberComponents.forEach((component, i) => {
            expect(component.prop("numberValue")).toBe(selectedNumbers[i])

        })
    })
    it("should call onNumberClick function when click triggered on selected Number element", () => {
        const onNumberClick = jest.fn()
        const selectedNumbers = [1, 2, 3, 4, 5, 6, 7, 8, 9]
        const wrapper = mount(<SelectedNumbersDisplay onNumberClick={onNumberClick}
                                                      selectedNumbers={selectedNumbers}/>)
        const number = wrapper.find(Number).first()
        number.simulate("click")
        expect(onNumberClick).toBeCalledTimes(1)
    })
});