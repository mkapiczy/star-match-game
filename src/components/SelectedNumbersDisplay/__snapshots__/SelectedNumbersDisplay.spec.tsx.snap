// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedNumbersDisplay component snapshot tests should match snapshot for all numbers selected 1`] = `
<div
  className="SelectedNumbersDisplay"
>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    9
  </button>
</div>
`;

exports[`SelectedNumbersDisplay component snapshot tests should match snapshot for no numbers selected 1`] = `
<div
  className="SelectedNumbersDisplay"
/>
`;

exports[`SelectedNumbersDisplay component snapshot tests should match snapshot for some numbers selected 1`] = `
<div
  className="SelectedNumbersDisplay"
>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    8
  </button>
</div>
`;
